### PR TITLE
14.0 mig product barcode constraint per company

### PR DESCRIPTION
**Pull request by slominchar**

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Thanks for porting this module. 
Could you add me as maintainer in the manifest ?

File: product_barcode_constraint_per_company/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -4,4 +4,4 @@ on the ``product.product`` model.
 With this module, two products can have the same barcode, if there don't belong
 to the same company.
 
-This contrainst was introduced in Odoo V9.0 in [that commit](https://github.com/odoo/odoo/commit/ede1071a09902e6eeb315cf51325a03f3991b3c3), and doesn't fit with some multi companies implementation where products are not shared.
+This contraint was introduced in Odoo V9.0 in [that commit](https://github.com/odoo/odoo/commit/ede1071a09902e6eeb315cf51325a03f3991b3c3), and doesn't fit with some multi companies implementation where products are not shared.
```
Comment (robinkeunen) [inline product_barcode_constraint_per_company/readme/DESCRIPTION.rst]: Still a little typo here : `conStraint`.

File: product_barcode_constraint_per_company/models/product_product.py [hunk 2]
```diff
@@ -2,25 +2,24 @@
...
     def _auto_init(self):
         for i, constraint in enumerate(self._sql_constraints):
             if constraint[0] == "barcode_uniq":
                 self._sql_constraints[i] = (
                     "barcode_uniq",
-                    "unique(barcode, tmpl_company_id)", _(
-                        "A barcode can only be assigned to one"
-                        " product per company !")
+                    "unique(barcode, tmpl_company_id)",
+                    _("A barcode can only be assigned to one" " product per company !"),
```
Comment (robinkeunen) [inline product_barcode_constraint_per_company/models/product_product.py]: Can you "unslplit" the string here ?

**Review by robinkeunen — APPROVED** [review]
A few cosmetic comments, otherwise it looks good 👍

File: product_barcode_constraint_per_company/models/product_product.py [hunk 3]
```diff
@@ -17,9 +17,10 @@ class ProductProduct(models.Model):
     def _auto_init(self):
         for i, constraint in enumerate(self._sql_constraints):
             if constraint[0] == "barcode_uniq":
+                self = self.with_context(prefetch_fields=False)
```
Comment (rousseldenis) [inline product_barcode_constraint_per_company/models/product_product.py]: You are losing context on each loop iteration.

I would have done it before loop.

Moreover, you cannot shadow 'self' name. Use a new variable name like 'self_without_prefetch' or ...

@slominchar Is it wanted to have that context in super() or just for setting _sql_constraints ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by slominchar** [comment]
Good morning @rousseldenis There is a strange issue when he try to run the runboat and fail with a field supostly he create installing purchase module

**Comment by rousseldenis** [comment]
> Good morning @rousseldenis There is a strange issue when he try to run the runboat and fail with a field supostly he create installing purchase module

Mmmh, could be a problem of dependencies

**Pull request merged**